nav2_mppi_controller/CMakeLists.txt

```diff
@@ -65,63 +53,126 @@ if(COMPILER_SUPPORTS_FMA)
 endif()
 
 # If building one the same hardware to be deployed on, try `-march=native`!
-add_compile_options(-O3 -finline-limit=10000000 -ffp-contract=fast -ffast-math -mtune=generic)
 
 add_library(mppi_controller SHARED
   src/controller.cpp
-  src/optimizer.cpp
   src/critic_manager.cpp
-  src/trajectory_visualizer.cpp
-  src/path_handler.cpp
-  src/parameters_handler.cpp
   src/noise_generator.cpp
+  src/optimizer.cpp
+  src/parameters_handler.cpp
+  src/path_handler.cpp
+  src/trajectory_visualizer.cpp
+)
+target_compile_options(mppi_controller PUBLIC -fconcepts -O3 -finline-limit=10000000 -ffp-contract=fast -ffast-math -mtune=generic)
```

Comment:
Added `-fconcepts`? We should be on C++17 which doesn't need that for the `auto` issue

Reply:
`-fconcepts` was already there, from the "old" line 97.  That said, I can remove it if you think we don't need it anymore.